### PR TITLE
SNOW-833259: Skip async doc test

### PR DESCRIPTION
**Pull request by sfc-gh-vnayak**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/async_job.py [hunk 1]
```diff
@@ -147,15 +147,15 @@ class AsyncJob:
             >>> from snowflake.snowpark.functions import col
             >>> query_id = session.sql("select 1 as A, 2 as B, 3 as C").collect_nowait().query_id
             >>> async_job = session.create_async_job(query_id)
-            >>> async_job.query
+            >>> async_job.query # doctest: +SKIP
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/async_job.py:150]: I have only seen this line failing, have the others all been flaky?

Comment (sfc-gh-vnayak) [reply src/snowflake/snowpark/async_job.py:150]: No, but in the cases where this line fails, the following lines do not get a chance to run (and do not have a chance to be surfaced as flaky lines). 

I reasoned that if the query test isn't working, it is likely that the result would be flaky as well.

Comment (sfc-gh-stan) [reply src/snowflake/snowpark/async_job.py:150]: I don't think the following lines are necessarily dependent on the first line. I prefer only skipping the ones that are shown to be flaky. cc: @sfc-gh-jdu

Comment (sfc-gh-vnayak) [reply src/snowflake/snowpark/async_job.py:150]: Sure, skipping only the two flaky lines.

**Review by sfc-gh-stan — APPROVED** [review]

**Pull request merged**